lib/riak/pool.ex

```diff
@@ -23,7 +23,12 @@ defmodule Riak.Pool do
         end
       end
       def unquote(name)(unquote_splicing(other_args)) do
-        pid = :pooler.take_group_member(:riak)
+        pid = case :pooler.take_group_member(:riak) do
+          :error_no_members ->
+            :timer.sleep(100 + :random.uniform(50)) # waits for a random time from 100ms to 150ms
```

Comment:
Sleeping a random amount of time is almost never the right answer, except when it is :-). Could you elaborate on how this changes the effect of the subsequent call to take_group_member? Also could you explain why we're confident that the subsequent call to take_group_member will not also return `:error_no_members`?

I would be curious to see what happens if we just immediately call take_group_member, vs sleeping a consistent amount of time (like 100ms), etc.

Another option here might be to implement our own take_group_member with a timeout:

```
...
  take_group_member(:riak, 500)
...

defp take_group_member(_, timeout) when timeout =< 0 do
  :error_no_members
end
defp take_group_member(group_name, timeout) do
  case :pooler.take_group_member(group_name) do
    :error_no_members ->
      #should probably log a warning / error here so the operator knows something is wrong
      :timer.sleep(100)
      take_group_member(group_name, timeout - 100)
    pid -> pid
  end
end